client/my-sites/upgrades/controller.jsx

```diff
@@ -216,7 +216,7 @@ module.exports = {
 
 		analytics.pageView.record( basePath, 'Checkout Thank You' );
 
-		context.store.dispatch( setSection( 'checkout-thank-you', { hasSidebar: false } ) );
+		context.store.dispatch( setSection( { name: 'checkout-thank-you' }, { hasSidebar: false } ) );
```

Comment:
I got confused here since I didn't get what we the function was doing. Now I think it's just an action creator function returning a action when is invoked. 

```json
{
  "type": "SECTION_SET",
  "name": "checkout-thank-you",
  "hasSidebar": false
}
```

I wonder how it could affect to the action since now we are passing an object instead of a string.
https://github.com/Automattic/wp-calypso/blob/master/client/state/ui/actions.js#L60

```es6
export function setSection( section, options = {} ) {
	options.type = SECTION_SET;
	if ( section ) {
		options.section = section;
	}
	options.hasSidebar = ( options.hasSidebar === false ) ? false : true;
	return options;
}
````

https://github.com/Automattic/wp-calypso/blob/master/client/state/ui/reducer.js#L64-L70

```es6
export function section( state = false, action ) {
	switch ( action.type ) {
		case SECTION_SET:
			return ( action.section !== undefined ) ? action.section : state;
	}
	return state;
}
```